src/anndata/_settings.py

```diff
@@ -134,9 +135,13 @@ def check_and_get_int(option, default_value):
 
 @dataclass
 class SettingsManager:
-    _registered_options: dict[str, RegisteredOption] = field(default_factory=dict)
-    _deprecated_options: dict[str, DeprecatedOption] = field(default_factory=dict)
-    _config: dict[str, object] = field(default_factory=dict)
+    _registered_options: OrderedDict[str, RegisteredOption] = field(
+        default_factory=OrderedDict
+    )
+    _deprecated_options: OrderedDict[str, DeprecatedOption] = field(
+        default_factory=OrderedDict
+    )
+    _config: OrderedDict[str, object] = field(default_factory=OrderedDict)
```

Comment:
why? Python dicts are guaranteed to maintain insertion order unless things get deleted. Do we delete things?